Seasons Defense/Assets/Scripts/Tower.cs

```diff
@@ -0,0 +1,29 @@
+using UnityEngine;
+
+public class Tower : MonoBehaviour
+{
+    public GameObject missilePrefab;
+    public float offset;
+    private Vector3 towerPosition;
+
+    private void Start()
+    {
+        towerPosition = new Vector3(transform.position.x, transform.position.y + offset, 0);
+    }
+
+    public void Fire() {
+
+        Vector3 mousePosition = new Vector3(Input.mousePosition.x, Input.mousePosition.y, transform.position.z);
+        Vector3 worldPosition = Camera.main.ScreenToWorldPoint(mousePosition);
+        worldPosition.z = transform.position.z;
+        GameObject missileInstance = Instantiate(missilePrefab, towerPosition, Quaternion.identity);
+      
+        // Give the missile the attributes it needs to move towards its target position
+        Missile missile = missileInstance.GetComponent<Missile>();
+
+        if (missile != null)
+        {
+            missile.target = worldPosition;
+        }
+    }
+}
```

Comment:
The ultimate request. Make sure to add a newline to the end of every script. GitHub and some low level languages will argue with it.
```suggestion
}

```